histomicstk/FeatureExtraction.py

```diff
@@ -129,6 +187,325 @@ def FeatureExtraction(Label, I, K=128, Fs=6, Delta=8):
+    df['CytoplasmEnergyGradMag'] = EosinGradientGroup.EnergyGradMag
+    df['CytoplasmSkewnessGradMag'] = EosinGradientGroup.SkewnessGradMag
+    df['CytoplasmKurtosisGradMag'] = EosinGradientGroup.KurtosisGradMag
+    df['CytoplasmSumCanny'] = EosinGradientGroup.SumCanny
+    df['CytoplasmMeanCanny'] = EosinGradientGroup.MeanCanny
+
+    df['Bounds'] = Bounds
 
     return df
+
+
+def GradientFeatureGroup(I, Coords):
+    """
+    Get GradientFeatures for nuclei and cytoplasms
+    """
+    Gx, Gy = np.gradient(I)
+    diffG = np.sqrt(Gx*Gx + Gy*Gy)
+    BW_canny = canny(I)
+
+    f = np.zeros((len(Coords), 8))
+    for i in range(len(Coords)):
+        pixOfInterest = diffG[Coords[i][:, 0], Coords[i][:, 1]]
+
+        f[i, 0] = np.mean(pixOfInterest)
+        f[i, 1] = np.std(pixOfInterest)
+        hist, bins = np.histogram(pixOfInterest, bins=np.arange(256))
+        prob = hist/np.sum(hist, dtype=np.float32)
+        f[i, 2] = entropy(pixOfInterest)
+        f[i, 3] = np.sum(np.power(prob, 2))
+        f[i, 4] = skew(pixOfInterest)
+        f[i, 5] = kurtosis(pixOfInterest)
+        bw_canny = BW_canny[Coords[i][:, 0], Coords[i][:, 1]]
+        f[i, 6] = np.sum(bw_canny)
+        f[i, 7] = f[i, 6] / len(pixOfInterest)
+
+    MeanGradMag = f[:, 0]
+    StdGradMag = f[:, 1]
+    EntropyGradMag = f[:, 2]
+    EnergyGradMag = f[:, 3]
+    SkewnessGradMag = f[:, 4]
+    KurtosisGradMag = f[:, 5]
+    SumCanny = f[:, 6]
+    MeanCanny = f[:, 7]
+
+    iFG = collections.namedtuple(
+        'iFG',
+        [
+            'MeanGradMag',
+            'StdGradMag',
+            'EntropyGradMag',
+            'EnergyGradMag',
+            'SkewnessGradMag',
+            'KurtosisGradMag',
+            'SumCanny',
+            'MeanCanny'
+        ]
+    )
+    Output = iFG(
+        MeanGradMag, StdGradMag, EntropyGradMag, EnergyGradMag,
+        SkewnessGradMag, KurtosisGradMag, SumCanny, MeanCanny
+    )
+
+    return Output
+
+
+def TextureFeatureGroup(I, Coords):
+    """
+    Get TextureFeatures for nuclei and cytoplasms
+    """
+    f = np.zeros((len(Coords), 4))
+    for i in range(len(Coords)):
+        pixOfInterest = I[Coords[i][:, 0], Coords[i][:, 1]]
+        hist, bins = np.histogram(pixOfInterest, bins=np.arange(256))
+        prob = hist/np.sum(hist, dtype=np.float32)
+        f[i, 0] = entropy(pixOfInterest)
+        f[i, 1] = np.sum(np.power(prob, 2))
+        f[i, 2] = skew(pixOfInterest)
+        f[i, 3] = kurtosis(pixOfInterest)
+
+    Entropy = f[:, 0]
+    Energy = f[:, 1]
+    Skewness = f[:, 2]
+    Kurtosis = f[:, 3]
+
+    iFG = collections.namedtuple(
+        'iFG',
+        [
+            'Entropy',
+            'Energy',
+            'Skewness',
+            'Kurtosis'
+        ]
+    )
+    Output = iFG(
+        Entropy, Energy, Skewness, Kurtosis
+    )
+
+    return Output
+
+
+def IntensityFeatureGroup(I, Coords):
+    """
+    Get IntensityFeatures for nuclei and cytoplasms
+    """
```

Comment:
I would advise you to get into the habit of first writing the docstring for the function before you start to write the logic. That way in your mind you will first think of 
* the inputs/parameters the function should take
* the outputs should return
* what the function does